32-bit windows support?

I want to build a 32-bit tauri app, but it failed with the following errors.
```
error[E0308]: mismatched types
    --> C:\Users\Liu.D.H\.cargo\registry\src\github.com-1285ae84e5963aae\tao-0.5.2\src\platform_impl\windows\window.rs:1051:20
     |
1051 |         userdata = createstruct.lpCreateParams as LONG_PTR;
     |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `i32`, found `isize`

For more information about this error, try `rustc --explain E0308`.
error: could not compile `tao` due to previous error
warning: build failed, waiting for other jobs to finish...
error: build failed
Error: failed to build app

Caused by:
    Result of `cargo build` operation was unsuccessful: exit code: 101

 app:spawn Command "D:\code\rust\tauri-app\node_modules\@tauri-apps\cli\bin\tauri-cli.exe" failed with exit code: 1 +0mserror Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

also https://github.com/tauri-apps/tao/issues/203.

I tried to update `src-tauri/Cargo.toml` according to the docs on https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html.
```
[package]
name = "app"
version = "0.1.0"
description = "A Tauri App"
authors = ["you"]
license = ""
repository = ""
default-run = "app"
edition = "2018"
build = "src/build.rs"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[build-dependencies]
tauri-build = { version = "1.0.0-beta.4" }

[dependencies]
serde_json = "1.0"
serde = { version = "1.0", features = ["derive"] }
tauri = { version = "1.0.0-beta.8", features = ["api-all"] }

[patch.crates-io]
tao = { git = "https://github.com/tauri-apps/tao", branch = "next" }

[features]
default = [ "custom-protocol" ]
custom-protocol = [ "tauri/custom-protocol" ]

```

Now I got some other errors.

<details>
<summary>`yarn tauri build`</summary>

```
D:\code\rust\tauri-app>yarn tauri build
yarn run v1.22.17
warning package.json: No license field
$ tauri build
 app:spawn Running "D:\code\rust\tauri-app\node_modules\@tauri-apps\cli\bin\tauri-cli.exe tauri build" +0ms

   Compiling winapi v0.3.9
   Compiling tao v0.5.2 (https://github.com/tauri-apps/tao?branch=next#dce2d529)
   Compiling windows v0.25.0
   Compiling app v0.1.0 (D:\code\rust\tauri-app\src-tauri)
   Compiling parking_lot_core v0.8.5
   Compiling webview2-sys v0.1.1
   Compiling winapi-util v0.1.5
   Compiling schannel v0.1.19
   Compiling dirs-sys-next v0.1.2
   Compiling remove_dir_all v0.5.3
   Compiling time v0.1.43
   Compiling filetime v0.2.15
   Compiling rfd v0.4.4
   Compiling shared_child v0.3.5
   Compiling os_info v3.0.8
   Compiling os_pipe v0.9.2
   Compiling open v2.0.2
   Compiling dirs-next v2.0.0
   Compiling tempfile v3.2.0
   Compiling same-file v1.0.6
   Compiling parking_lot v0.11.2
   Compiling tar v0.4.37
   Compiling native-tls v0.2.8
   Compiling zip v0.5.13
   Compiling webview2 v0.1.4
   Compiling walkdir v2.3.2
   Compiling string_cache v0.8.2
   Compiling attohttpc v0.17.0
   Compiling ignore v0.4.18
   Compiling markup5ever v0.10.1
   Compiling html5ever v0.25.1
   Compiling kuchiki v0.8.1
   Compiling tauri-utils v1.0.0-beta.3
   Compiling tauri-runtime v0.2.1
   Compiling wry v0.12.2
error[E0433]: failed to resolve: could not find `ole2` in `um`
  --> C:\Users\Liu.D.H\.cargo\registry\src\github.com-1285ae84e5963aae\wry-0.12.2\src\webview\webview2\file_drop.rs:72:22
   |
72 |       if winapi::um::ole2::RevokeDragDrop(hwnd) != winapi::shared::winerror::DRAGDROP_E_INVALIDHWND
   |                      ^^^^ could not find `ole2` in `um`

error[E0433]: failed to resolve: could not find `ole2` in `um`
  --> C:\Users\Liu.D.H\.cargo\registry\src\github.com-1285ae84e5963aae\wry-0.12.2\src\webview\webview2\file_drop.rs:73:24
   |
73 |         && winapi::um::ole2::RegisterDragDrop(hwnd, handler_interface_ptr) == S_OK
   |                        ^^^^ could not find `ole2` in `um`

error[E0308]: mismatched types
   --> C:\Users\Liu.D.H\.cargo\registry\src\github.com-1285ae84e5963aae\wry-0.12.2\src\webview\webview2\mod.rs:146:21
    |
145 |               let cursor = match result {
    |                                  ------ this expression has type `windows::Windows::Win32::Foundation::LRESULT`
146 |                     HTLEFT => CursorIcon::WResize,
    |                     ^^^^^^ expected struct `windows::Windows::Win32::Foundation::LRESULT`, found `isize`
    |
help: you might have meant to use field `0` whose type is `isize`
    |
145 |               let cursor = match result.0 {
    |                                  ~~~~~~~~

error[E0308]: mismatched types
   --> C:\Users\Liu.D.H\.cargo\registry\src\github.com-1285ae84e5963aae\wry-0.12.2\src\webview\webview2\mod.rs:147:21
    |
145 |               let cursor = match result {
    |                                  ------ this expression has type `windows::Windows::Win32::Foundation::LRESULT`
146 |                     HTLEFT => CursorIcon::WResize,
147 |                     HTTOP => CursorIcon::NResize,
    |                     ^^^^^ expected struct `windows::Windows::Win32::Foundation::LRESULT`, found `isize`
    |
help: you might have meant to use field `0` whose type is `isize`
    |
145 |               let cursor = match result.0 {
    |                                  ~~~~~~~~

error[E0308]: mismatched types
   --> C:\Users\Liu.D.H\.cargo\registry\src\github.com-1285ae84e5963aae\wry-0.12.2\src\webview\webview2\mod.rs:148:21
    |
145 |               let cursor = match result {
    |                                  ------ this expression has type `windows::Windows::Win32::Foundation::LRESULT`
...
148 |                     HTRIGHT => CursorIcon::EResize,
    |                     ^^^^^^^ expected struct `windows::Windows::Win32::Foundation::LRESULT`, found `isize`
    |
help: you might have meant to use field `0` whose type is `isize`
    |
145 |               let cursor = match result.0 {
    |                                  ~~~~~~~~

error[E0308]: mismatched types
   --> C:\Users\Liu.D.H\.cargo\registry\src\github.com-1285ae84e5963aae\wry-0.12.2\src\webview\webview2\mod.rs:149:21
    |
145 |               let cursor = match result {
    |                                  ------ this expression has type `windows::Windows::Win32::Foundation::LRESULT`
...
149 |                     HTBOTTOM => CursorIcon::SResize,
    |                     ^^^^^^^^ expected struct `windows::Windows::Win32::Foundation::LRESULT`, found `isize`
    |
help: you might have meant to use field `0` whose type is `isize`
    |
145 |               let cursor = match result.0 {
    |                                  ~~~~~~~~

error[E0308]: mismatched types
   --> C:\Users\Liu.D.H\.cargo\registry\src\github.com-1285ae84e5963aae\wry-0.12.2\src\webview\webview2\mod.rs:150:21
    |
145 |               let cursor = match result {
    |                                  ------ this expression has type `windows::Windows::Win32::Foundation::LRESULT`
...
150 |                     HTTOPLEFT => CursorIcon::NwResize,
    |                     ^^^^^^^^^ expected struct `windows::Windows::Win32::Foundation::LRESULT`, found `isize`
    |
help: you might have meant to use field `0` whose type is `isize`
    |
145 |               let cursor = match result.0 {
    |                                  ~~~~~~~~

error[E0308]: mismatched types
   --> C:\Users\Liu.D.H\.cargo\registry\src\github.com-1285ae84e5963aae\wry-0.12.2\src\webview\webview2\mod.rs:151:21
    |
145 |               let cursor = match result {
    |                                  ------ this expression has type `windows::Windows::Win32::Foundation::LRESULT`
...
151 |                     HTTOPRIGHT => CursorIcon::NeResize,
    |                     ^^^^^^^^^^ expected struct `windows::Windows::Win32::Foundation::LRESULT`, found `isize`
    |
help: you might have meant to use field `0` whose type is `isize`
    |
145 |               let cursor = match result.0 {
    |                                  ~~~~~~~~

error[E0308]: mismatched types
   --> C:\Users\Liu.D.H\.cargo\registry\src\github.com-1285ae84e5963aae\wry-0.12.2\src\webview\webview2\mod.rs:152:21
    |
145 |               let cursor = match result {
    |                                  ------ this expression has type `windows::Windows::Win32::Foundation::LRESULT`
...
152 |                     HTBOTTOMLEFT => CursorIcon::SwResize,
    |                     ^^^^^^^^^^^^ expected struct `windows::Windows::Win32::Foundation::LRESULT`, found `isize`
    |
help: you might have meant to use field `0` whose type is `isize`
    |
145 |               let cursor = match result.0 {
    |                                  ~~~~~~~~

error[E0308]: mismatched types
   --> C:\Users\Liu.D.H\.cargo\registry\src\github.com-1285ae84e5963aae\wry-0.12.2\src\webview\webview2\mod.rs:153:21
    |
145 |               let cursor = match result {
    |                                  ------ this expression has type `windows::Windows::Win32::Foundation::LRESULT`
...
153 |                     HTBOTTOMRIGHT => CursorIcon::SeResize,
    |                     ^^^^^^^^^^^^^ expected struct `windows::Windows::Win32::Foundation::LRESULT`, found `isize`
    |
help: you might have meant to use field `0` whose type is `isize`
    |
145 |               let cursor = match result.0 {
    |                                  ~~~~~~~~

error[E0308]: mismatched types
   --> C:\Users\Liu.D.H\.cargo\registry\src\github.com-1285ae84e5963aae\wry-0.12.2\src\webview\webview2\mod.rs:164:30
    |
164 |                 if result != HTCLIENT {
    |                              ^^^^^^^^ expected struct `windows::Windows::Win32::Foundation::LRESULT`, found `isize`

error[E0061]: this function takes 4 arguments but 1 argument was supplied
   --> C:\Users\Liu.D.H\.cargo\registry\src\github.com-1285ae84e5963aae\wry-0.12.2\src\webview\webview2\mod.rs:165:27
    |
165 |                   window_.begin_resize_drag(result);
    |                           ^^^^^^^^^^^^^^^^^ ------ supplied 1 argument
    |                           |
    |                           expected 4 arguments
    |
note: associated function defined here
   --> C:\Users\Liu.D.H\.cargo\git\checkouts\tao-b2c91e21b396bd53\dce2d52\src\platform\windows.rs:114:6
    |
114 |   fn begin_resize_drag(&self, edge: isize, button: u32, x: i32, y: i32);
    |      ^^^^^^^^^^^^^^^^^

error[E0605]: non-primitive cast: `*mut c_void` as `windows::Windows::Win32::Foundation::HWND`
   --> C:\Users\Liu.D.H\.cargo\registry\src\github.com-1285ae84e5963aae\wry-0.12.2\src\webview\webview2\mod.rs:144:37
    |
144 |               let result = hit_test(window_.hwnd() as _, cx, cy);
    |                                     ^^^^^^^^^^^^^^^^^^^ an `as` expression can only be used to convert between primitive types or to coerce to a specific trait object

Some errors have detailed explanations: E0061, E0308, E0433, E0605.
For more information about an error, try `rustc --explain E0061`.
error: could not compile `wry` due to 13 previous errors
warning: build failed, waiting for other jobs to finish...
error: build failed
Error: failed to build app

Caused by:
    Result of `cargo build` operation was unsuccessful: exit code: 101

 app:spawn Command "D:\code\rust\tauri-app\node_modules\@tauri-apps\cli\bin\tauri-cli.exe" failed with exit code: 1 +0ms
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

D:\code\rust\tauri-app>
```
</details>

--- a/.changes/cli.rs-fix-windows-x86.md
+++ b/.changes/cli.rs-fix-windows-x86.md
@@ -1,0 +1,5 @@
+---
+"cli.rs": patch
+---
+
+Fix `build` command when executed on a 32-bit Windows machine when pulling from the `binary-releases` repo.

--- a/tooling/cli.rs/src/build.rs
+++ b/tooling/cli.rs/src/build.rs
@@ -174,22 +174,17 @@ pub fn command(options: Options) -> Result<()> {
     // move merge modules to the out dir so the bundler can load it
     #[cfg(windows)]
     {
-      let arch = if let Some(t) = &options.target {
-        if t.starts_with("x86_64") {
-          "x86_64"
-        } else if t.starts_with('i') {
-          "x86"
-        } else if t.starts_with("arm") {
-          "arm"
-        } else if t.starts_with("aarch64") {
-          "aarch64"
-        } else {
-          panic!("Unexpected target triple {}", t)
-        }
-      } else if cfg!(target_arch = "x86") {
-        "x86"
-      } else {
+      let target = options.target.clone().unwrap_or_else(|| std::env::consts::ARCH.into());
+      let arch = if target.starts_with("x86_64") {
         "x86_64"
+      } else if target.starts_with('i') || target.starts_with("x86") {
+        "x86"
+      } else if target.starts_with("arm") {
+        "arm"
+      } else if target.starts_with("aarch64") {
+        "aarch64"
+      } else {
+        panic!("Unexpected target architecture {}", target.split("_").next().unwrap())
       };
       let (filename, vcruntime_msm) = if arch == "x86" {
         let _ = std::fs::remove_file(out_dir.join("Microsoft_VC142_CRT_x64.msm"));

